src/learner.cc

```diff
@@ -731,6 +731,12 @@ class LearnerIO : public LearnerConfiguration {
       tparam_.dsplit = DataSplitMode::kRow;
     }
 
+    // Do not load "process_type":"update", this causes all trees to be removed
+    // from the model on configuration
+    if (cfg_.find("process_type") == cfg_.end()) {
```

Comment:
I'm not comfortable with putting this in Learner.  
- `process_type` is a training parameter and is not a model parameter, which means you don't need this parameter to define what a model is or how it looks like.  So the hack should go into `LoadConfig`.
- `Load` and `Save` should produce exactly the same internal states, only this way can we guarantee a pickle thrown to another worker or recovered from failure during training can continue to work.
- Also `process_type` is a gbtree specific parameter.